packages/next/client/route-loader.ts

```diff
@@ -94,7 +94,10 @@ function prefetchViaDom(
   link?: HTMLLinkElement
 ): Promise<any> {
   return new Promise<void>((res, rej) => {
-    if (document.querySelector(`link[rel="prefetch"][href^="${href}"]`)) {
+    const selector = `
+      link[rel="prefetch"][href^="${href}"],
+      link[rel="preload"][href^="${href}"]`
```

Comment:
```suggestion
      link[rel="preload"][href^="${href}"],
      script[src^="${href}"]``
```
It looks like this doesn't resolve the issue since we use script tags with the defer prop instead of preloads now. We can leave the preload check here incase the optimized loading is disabled though.